{{cookiecutter.profile_name}}/OSLayer.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+import subprocess
+from typing import Tuple
+import random
+import string
+
+
+class OSLayer:
+    """
+    This class provides an abstract layer to communicating with the OS.
+    Its main purpose is to enable OS operations mocking, so we don't actually need to make file operations or create
+    processes.
+    """
+    @staticmethod
+    def mkdir(directory: Path):
+        directory.mkdir(parents=True, exist_ok=True)
+
+    @staticmethod
+    def remove_file(file: Path):
+        if file.exists():
```

Comment:
It would be more "correct" to make this check 

```py
if file.is_file():
    file.unlink()
```
`is_file()` includes a check for `exists()` and saves us in the case where `file` is a directory, as `unlink()` does not work on directories.

Reply:
Done in https://github.com/leoisl/snakemake-lsf/commit/5155b82cf3d3b2b87cbcd6de31ad28a4fe6766e0